sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/EventHubsTestEnvironment.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Testing;
+using Azure.Messaging.EventHubs.Core;
+
+namespace Azure.Messaging.EventHubs.Tests
+{
+    /// <summary>
+    ///   Represents the ambient environment in which the test suite is
+    ///   being run, offering access to information such as environment
+    ///   variables.
+    /// </summary>
+    ///
+    public class EventHubsTestEnvironment: TestEnvironment
+    {
+        public EventHubsTestEnvironment() : base("eventhub")
+        {
+            ParsedConnectionString = new Lazy<ConnectionStringProperties>(() => ConnectionStringParser.Parse(EventHubsConnectionString), LazyThreadSafetyMode.ExecutionAndPublication);
+            ActiveEventHubsNamespace = new Lazy<NamespaceProperties>(EnsureEventHubsNamespace, LazyThreadSafetyMode.ExecutionAndPublication);
+        }
+
+        public static EventHubsTestEnvironment Instance { get; } = new EventHubsTestEnvironment();
```

Comment:
Please keep the order of  readonly and const fields before property definitions.